nopassword/models.py

```diff
@@ -1,64 +1,30 @@
 # -*- coding: utf-8 -*-
 import hashlib
 import os
-from datetime import datetime
 
 from django.conf import settings
-from django.contrib.auth import get_backends
-from django.core.urlresolvers import reverse_lazy
 from django.db import models
 from django.utils import timezone
 from django.utils.translation import ugettext_lazy as _
 
-from .utils import AUTH_USER_MODEL, get_username
-
 
 class LoginCode(models.Model):
-    user = models.ForeignKey(AUTH_USER_MODEL, related_name='login_codes',
-                             editable=False, verbose_name=_('user'))
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, related_name='login_codes',
+                             editable=False, verbose_name=_('user'), on_delete=models.CASCADE)
     code = models.CharField(max_length=20, editable=False, verbose_name=_('code'))
     timestamp = models.DateTimeField(editable=False)
     next = models.TextField(editable=False, blank=True)
 
-    def __unicode__(self):
+    def __str__(self):
         return "%s - %s" % (self.user, self.timestamp)
 
     def save(self, *args, **kwargs):
-        if settings.USE_TZ:
-            self.timestamp = timezone.now()
-        else:
-            self.timestamp = datetime.now()
+        self.timestamp = timezone.now()
 
         if not self.next:
             self.next = '/'
-        super(LoginCode, self).save(*args, **kwargs)
 
-    def login_url(self, secure=False, host=None):
```

Comment:
Backward incompatible: Moved to `forms` so that one has more control over the used url.